### ISSUE TITLE
Fix 500 error when getting a non-existing project

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#900]
A change that came about with `1.3.12a0` is that the creation of a project does not automatically also create a dataset associated to it.
While testing in fractal-web, I found the following 500 error. I think the way to reproduce it is to GET a non-existing project.

```python

INFO:     127.0.0.1:52984 - "GET /api/v1/project/1 HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/applications.py", line 292, in __call__
    await super().__call__(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/routing.py", line 291, in app
    content = await serialize_response(
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/routing.py", line 154, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 1 validation errors:
  {'loc': ('response', 'dataset_list', 0, 'history'), 'msg': 'none is not an allowed value', 'type': 'type_error.none.not_allowed'}

**Comment by tcompa** [comment]
I cannot reproduce this error with a fresh db.
I will check again, trying to first create a DB with 1.3.11 and then migrating to 1.3.12a0.

**Comment by tcompa** [comment]
Here is how to reproduce the error:
* Install fractal-server 1.3.11
* Run set-db and start
* Create a project
* Stop fractal-server
* Install fractal-server 1.3.12a0
* Run set-db
```
$ fractalctl set-db
Run alembic.config, with argv=['-c', '/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fractal_server/alembic.ini', 'upgrade', 'head']
fractal_server.app.db
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 8f79bd162e35 -> 7a5a07816dd5, Add Dataset.history
```
* GET projects
```
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     127.0.0.1:57524 - "GET /auth/whoami HTTP/1.1" 200 OK
INFO:     127.0.0.1:57538 - "GET /api/alive/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:57524 - "GET /auth/whoami HTTP/1.1" 200 OK
INFO:     127.0.0.1:57546 - "GET /api/v1/project/ HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/applications.py", line 292, in __call__
    await super().__call__(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/routing.py", line 291, in app
    content = await serialize_response(
  File "/home/tommaso/Fractal/fractal-web/lib/fractal-server/venv/lib/python3.10/site-packages/fastapi/routing.py", line 154, in serialize_response
    raise ResponseValidationError(
fastapi.exceptions.ResponseValidationError: 1 validation errors:
  {'loc': ('response', 0, 'dataset_list', 0, 'history'), 'msg': 'none is not an allowed value', 'type': 'type_error.none.not_allowed'}
```

**Comment by tcompa** [comment]
TLDR, based on long test session with @mfranzon 

We are still unable to set a new column value to some JSON value (e.g. `[]`). The defaults of `sqlmodel.Field` do matter only when a new object is created, and do not enter in the migration. Modifying the migration script as in
```python
  def upgrade() -> None:
      with op.batch_alter_table("dataset", schema=None) as batch_op:
        batch_op.add_column(sa.Column("history", sa.JSON(), nullable=True, default=list))
```
does not solve the problem.

Likely we will have to make the attribute optional.

**Comment by tcompa** [comment]
Implemented via [1ea7968](https://github.com/fractal-analytics-platform/fractal-server/commit/1ea79687ce9270377b3ba5ccfac2be04d271f749).
Notice also the use of `server_default`:
```python
sa_column=Column(JSON, server_default="[]", nullable=False)
```
